docs/rfc/rfc-8.adoc

```diff
@@ -0,0 +1,482 @@
+:toc: macro
+
+= RFC 8: Cross-chain Tokenized Threshold BTC
+
+:icons: font
+:numbered:
+toc::[]
+
+== Goal
+
+The goal of this proposal is to enable *efficient* cross-chain tBTC liquidity.
+
+In the Ethereum ecosystem, that means bringing Bitcoin to L2s and sidechains like
+Arbitrum, Optimism, Polygon, Avalanche, and Binance Smart Chain. In the Cosmos
+ecosystem, that means Bitcoin on IBC-enabled zones (chains) like Osmosis and
+Penumbra. And in newer ecosystems, that means chains like Solana, Aptos, and
+Sui.
+
+Efficiency requires minimizing liquidity fragmentation by avoiding the need for
+tBTC wrapper tokens by each chain and bridge. This situation quickly leads to an
+`n*m` problem, where liquidity on `n` chains is divided between `m`
+bridges. Each of these token variants soaks up valuable liquidity that could be
+better utilized elsehwere.
+
+Because tBTC is minted "centrally" on Ethereum L1, we can do better. By
+deploying a canonical token on each chain, we can ensure the supply of tBTC
+remains sacrosanct, while enabling quick, interoperable cross-chain bridges and
+localizing ecosystem risk.
+
+== Overview
+
+To achieve this goal, I propose we deploy a canonical tBTC contract on each
+chain we support. This contract should be flexible enough to:
+
+* Delegate restricted minting authority to a native bridge on the chain, if
+  present.
+* Delegate restricted minting authority to a short list of ecosystem bridges
+* Be paused by any one of `n` guardians, allowing avoidance of contagion in case
+  of a chain- or bridge-specific incident.
+* Be governed by the Threshold Council until we can land on a longer-term
+  cross-chain governance mechanism.
+
+== Initial implementation
+
+The goal of the initial implementation is to bring Bitcoin to the following L2s
+and sidechains:
+
+* Arbitrum,
+* Polygon,
+* Optimism.
+
+The cross-bridge architecture and patterns we are going to choose should allow
+us to scale to other L2s, sidechains, and chains quickly.
+
+=== Minting on L1
+
+Rather than change our L1 implementation to be aware of L2s and sidechains,
+we'll start as simply as possible. Every tBTC mint will continue to require
+transactions on Ethereum from the user — first a deposit reveal, then a lock
+to bridge to another network.
+
+=== Canonical token
+
+The canonical token we are going to deploy will be different from the L1 token
+deployed at `0x18084fbA666a33d37592fA2633fD49a74DD93a88`. The L2/sidechain token
+will be upgradeable, owned by Threshold Council, will delegate the minting
+authority to multiple parties, and have a pause functionality for mints and
+burns.
+
+Given those requirements, the address of the canonical tBTC token on L2s and
+sidechains will be different from the one deployed on L1.
+
+A generic implementation of the token will be provided in the `solidity` smart
+contracts directory.
```

Comment:
Do you mean this repository https://github.com/keep-network/tbtc-v2/tree/main/solidity?
Could we consider placing it in a separate package, so we don't need to install [all the dependencies](https://github.com/keep-network/tbtc-v2/blob/24b0b974bd14d576806cf57c932ade1ca1041aa4/solidity/package.json#L33-L75) of `@keep-network/tbtc-v2`?

Perhaps https://github.com/thesis/solidity-contracts would be the cleanest place to hold it? Or a dedicated subdirectory in `tbtc-v2` repository with an independent NPM package?

Reply:
I would go with `keep-network/tbtc-v2/solidity` for simplicity. We would need to import this repository and all transitive dependencies for L1 TBTC token artifact anyway.